### PR TITLE
feat[dcnm_policy]: adding the functions to handle use_desc_as_key

### PR DESCRIPTION
**Pull request by dsx1123**
create the PR to track the changes. Will add test cases and update the document

### CODE REVIEW

**Comment by dsx1123** [comment]
#279

File: plugins/modules/dcnm_policy.py [hunk 1]
```diff
@@ -44,6 +44,13 @@
       - query
     default: merged
 
+  use_desc_as_key:
+    description:
+    - A flag specifying whether using the description as unique key for policies.
```
Comment (mikewiebe) [inline plugins/modules/dcnm_policy.py:49]: Suggest rewording this a bit

`Flag to enforce using the description parameter as the unique key for policy management"

File: plugins/modules/dcnm_policy.py [hunk 2]
```diff
@@ -715,7 +762,11 @@ def dcnm_policy_get_diff_merge(self):
 
             rc, policy_id = self.dcnm_policy_compare_policies(policy)
 
-            if rc == "DCNM_POLICY_ADD_NEW":
+            if rc == "DCNM_POLICY_DUPLICATED":
+                self.module.fail_json(
+                    f"Policies with the same description are found in DCNM/NDFC: {self.use_desc_as_key}, {policy['description']}"
```
Comment (mikewiebe) [inline plugins/modules/dcnm_policy.py:767]: Suggest adding this at the beginning 

`f"Multiple policies found with the same description in DCNM/NDFC...`

Just makes it a bit more clear, since one is ok for modification

File: plugins/modules/dcnm_policy.py (hunk 1 above)
Comment (dsx1123) [reply plugins/modules/dcnm_policy.py]: Done

File: plugins/modules/dcnm_policy.py (hunk 2 above)
Comment (dsx1123) [reply plugins/modules/dcnm_policy.py]: Done

File: plugins/modules/dcnm_policy.py [hunk 3]
```diff
@@ -338,6 +349,46 @@
...
       - switch:
           - ip: "{{ ansible_switch1 }}"
+
+# Use the description as key
+
+# NOTE: As the description of the policy in NDFC/DCNM is not unique,
+#       the user must make sure no policies with the same description are created on NDFC out of the playbook.
+#       If the description is not unique, the module will raise an error.
+
+## Below task will policies with description "policy_radius" on swtich1, switch2 and switch3,
```
Comment (mmudigon) [inline plugins/modules/dcnm_policy.py:359]: typo, should read as

Below task will create policies ...

File: plugins/modules/dcnm_policy.py [hunk 4]
```diff
@@ -637,12 +716,20 @@ def dcnm_policy_get_have(self):
             pl
             for pl in plist
             for wp in self.want
-            if (pl["templateName"] == wp["templateName"])
+            if pl.get("source", "") == ""
```
Comment (mmudigon) [inline plugins/modules/dcnm_policy.py:719]: What is this check for? Not clear. Can you please add a comment for pl.get("source", "")?

File: plugins/modules/dcnm_policy.py [hunk 5]
```diff
@@ -1251,7 +1437,7 @@ def dcnm_translate_config(self, config):
                     updated_config.append(cfg)
             config = updated_config
         else:
-            config = override_config
+            config += override_config
```
Comment (mmudigon) [inline plugins/modules/dcnm_policy.py:1440]: Is this working? When I checked it said '+=' not supported for dicts. Suggested to use update instead.

File: plugins/modules/dcnm_policy.py [hunk 6]
```diff
@@ -943,6 +1098,35 @@ def dcnm_policy_create_policy(self, policy, command):
 
         return resp
 
+    def dcnm_policy_update_policy(self, policy, command):
```
[the remark this answers is not in the document]
Comment (dsx1123) [reply plugins/modules/dcnm_policy.py:1101]: I have tried the PUT method using `POLICY_BULK_CREATE` URL, it didn't work, besides, it is not documented. So prefer to use the documented endpoint for the update.

File: plugins/modules/dcnm_policy.py (hunk 3 above)
Comment (dsx1123) [reply plugins/modules/dcnm_policy.py:359]: will fix it.

File: plugins/modules/dcnm_policy.py (hunk 5 above)
Comment (dsx1123) [reply plugins/modules/dcnm_policy.py:1440]: `config` and `override_config` are lists, not dicts, and yes, it works

**Comment by mikewiebe** [comment]
@dsx1123 and @mmudigon is there anything pending for this PR?

**Comment by dsx1123** [comment]
> @dsx1123 and @mmudigon is there anything pending for this PR?

I should have addressed all the comments, but pending review again from @mmudigon

**Review by mmudigon — APPROVED** [review]

**Review by mikewiebe — APPROVED** [review]
